pkgs/tools/typesetting/tex/texlive/combine.nix

```diff
@@ -132,18 +139,30 @@ in (buildEnv {
         + lib.concatMapStrings (pname: section "^-- from ${pname}:$" "^}$|^-- from") hyphenPNames
         + "$p;\n"
       );
+    # sed expression that enables the lines in /start/,/end/
+    activateFormats = pname: "/^# from ${pname}:$/,/^# from/{ s/^#! //; };\n";
     fmtutilSed =
       writeText "fmtutil.sed" (
-        "1{ s/^(# Generated by .*)$/\\1, modified by texlive.combine/; p; }\n"
-        + "2,/^# from/{ /^# from/!p; };\n"
-        + lib.concatMapStrings (pname: section "^# from ${pname}:$" "^# from") formatPNames
+        # document how file was generated
+        "1{ s/^(# Generated by .*)$/\\1, modified by texlive.combine/; }\n"
+        # disable all formats, even those already disabled
+        + "s/^([^#]|#! )/#! \\1/;\n"
+        # enable the requested formats
+        + lib.concatMapStrings activateFormats formatPNames
+        # clean up formats that have been disabled twice
+        + "s/^#! #! /#! /;\n"
```

Comment:
So this is where it gets tricky, because both files in `TEXMFDIST` and `TEXMFSYSVAR` are read by `fmutil`. The old combine could simply delete the missing formats; now, we cannot delete them from `TEXMFDIST`, thus we must *disable* them by prepending `#!` (as per documentation [1]).

This required some fiddling with `fmtutil.sed` about adding `#!`, then removing it for the packages that are being installed, and finally cleaning up the stray double `#!`'s caused by formats that are already disabled in the CTAN file (to be fair, the clean up is probably redundant). I hope the code is not too obscure.

[1] Section "Disabling formats" https://manpages.debian.org/experimental/texlive-base/fmtutil.1.en.html

Reply:
Added an extra documentation line referring to `man fmtutil`.